enhancements/update/cvo-log-level-api.md

```diff
@@ -0,0 +1,451 @@
+Given a cluster administrator and a working standalone cluster for which the 
+administrator is responsible.
+
+**Cluster administrator** is a human user responsible for managing the cluster.
+
+1. The cluster administrator notices an issue in the cluster and chooses to
+   troubleshoot the issue.
+2. The cluster administrator, after some troubleshooting, notices that the logs
+   of the Cluster Version Operator (CVO) might help.
+3. The cluster administrator notices that the logs are not detailed enough to
+   troubleshoot the issue.
+4. The cluster administrator raises the log level from the default value to a
+   more verbose level by simply modifying the new `ClusterVersionOperator`
+   resource named `cluster` via the web console or by patching the resource by
+   using the CLI.
+5. The cluster administrator fixes the issue in the cluster.
+6. The cluster administrator notices that the CVO outputs too many logs for the
+   administrator's liking.
+7. The cluster administrator lowers the log level of the CVO to the lowest 
+   level, the default level.
+8. The cluster administrator is now a happy cluster administrator.
+
+### API Extensions
+
+The enhancement proposes to create a new `ClusterVersionOperator` CRD. A new
+`ClusterVersionOperator` resource will only impact the CVO logging level.
+
+In HyperShift, its API and logic will be slightly extended.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+A hosted CVO is located in the management cluster and accesses the hosted API
+server. Applying the configuration in the hosted cluster would enable a hosted
+cluster administrator to modify a workload running in the management cluster.
+A hosted cluster administrator could set the log level of the hosted CVO in the
+management cluster, thus affecting the storage space of the management cluster.
+
+Thus, the new CRD and the CR will not be applied in a hosted cluster. This 
+will be achieved using the [cluster profiles][cluster-profiles]. The new CRD 
+and CR will be missing the `include.release.openshift.io/hypershift: "true"` 
+annotation, and thus they won't be applied in a hosted cluster.
+
+In HyperShift, the `HostedCluster` API will be extended to include a CVO 
+configuration. This configuration will be then transferred to a 
+configuration file that will be accessible to the hosted CVO. Meaning that 
+the HyperShift will generate a configuration file for the hosted CVO based 
+on the set HyperShift API values. The hosted CVO, when configured to use a 
+configuration file, will not try to reconcile a configuration CR using the API 
+server.
+
+This way no CRDs will be created in the management cluster, a problem since 
+a management cluster may host multiple OCP clusters of different versions. No 
+configuration API will be introduced in the hosted cluster that would have to be
+otherwise protected by a validating admission policy.
+
+The configuration file may be a manifest file of a `ClusterVersionOperator` 
```

Comment:
It's a GREAT option, and it will be implemented in such a way with the probability of 99%. My thinking was to not write in stone the definitive implementation details of such a matter.

However, now that I think about it, there is no need to leave the space open, as the option was openly discussed and was agreed upon in this enhancement. There is no need. I have updated the wording. In case we discover a greatly better alternative while developing, we can always circle back.

> I guess the one issue is if the hosted version needs a new configuration that the standalone version does not

My thinking was we could then create a new API, `HostedClusterVersionOperator`. Its CRD would not be applied to any cluster. The hosted CVO could then load a manifest file of this kind. The CVO could then differentiate the configuration files based on the `GroupVersionKind` and would act accordingly. Something along these lines. A new API to communicate between components while not being exposed to users.